jaxlib/lapack.pyx

```diff
@@ -77,7 +78,11 @@ cdef void blas_strsm(void* out, void** data) nogil:
   cdef char cdiag = 'U' if diag else 'N'
   cdef int lda = m if left_side else n
   cdef int ldb = m
-  strsm(&cside, &cuplo, &ctransa, &cdiag, &m, &n, alpha, a, &lda, x, &ldb)
+
+  for _ in range(num_b):
+    strsm(&cside, &cuplo, &ctransa, &cdiag, &m, &n, alpha, a, &lda, x, &ldb)
+    x += n*m
+    a += lda**2
```

Comment:
I think you should cast `lda` to 64 bit, and compute this as `<int64_t>(lda) * <int64_t>(lda)`.

I am not completely sure of Cython semantics, but I suspect`** 2` will perform its calculation as a floating point type.

Same everywhere.

Reply:
lda refers to the dimensional size of a, so either `m` or `n`. Happy to cast this up if you decide the `m` and `n` should be set to 64-bit.